docs/pages/blog/callback-support-in-style-overrides.md

```diff
@@ -0,0 +1,189 @@
+          root: ({ ownerState, theme }) => ({
+            padding: {
+              small: '8px 4px',
+              medium: '12px 6px',
+              large: '16px 8px',
+            }[ownerState.size],
+            ...(ownerState.variant === 'outlined' && {
+              borderWidth: '2px',
+              ...(ownerState.variant === 'primary' && {
+                borderColor: theme.palette.primary.light,
+              }),
+            }),
+          }),
+          label: {
+            padding: 0,
+          },
+        },
+      },
+    },
+  })}
+>
+  ...your app
+</ThemeProvider>;
+```
+
+> 💡 The side benefit of using a callback is that you can use the runtime theme without creating the outer scoped variable.
+
+### TypeScript
+
+The callback is type-safe.
+
+- `ownerState`: `ComponentProps` interface, eg. `ButtonProps`, `ChipProps`, etc.
+- `theme`: `Theme` interface from `@mui/material/styles`.
+
+```tsx
+{
+  MuiChip: {
+    styleOverrides: {
+      // ownerState: ChipProps
+      // theme: Theme
+      root: ({ ownerState, theme }) => ({...}),
+    },
+  }
+}
+```
+
+If you extend the interface via module augmentation like this:
+
+```ts
+declare module '@mui/material/Button' {
+  interface ButtonPropsVariantOverrides {
+    dashed: true;
+  }
+}
+```
+
+you will be able to see those props in `ownerState.variant` 🎉. `theme` can be augmented as well.
+
+## Experimental `sx` function
+
+Initially, `sx` was designed to be a prop that enables you to inject styles with a shorthand notation to components created with the `styled` API:
+
+```jsx
+import { styled } from '@mui/material/styles';
+import Box from '@mui/material/Box';
+
+const Label = styled('span')({
+  fontWeight: 'bold',
+  fontSize: '0.875rem',
+})
+
+<Box sx={{ display: 'flex' }}>
+  <Label sx={{ color: 'text.secondary' }}>Label</Label>
+</Box>;
+```
+
+> 💡 All MUI components are created with `styled` API, so they accept `sx` prop by default.
+
+`sx` helps developers write less code and be more productive once they are familiar with the API. With the callback support in `styleOverrides`, it is now possible to use `sx` like syntax at the global theme overrides.
+
+All you need is to use the [`experimental_sx`](/system/styled/#how-can-i-use-the-sx-syntax-with-the-styled-utility) function:
+
+```jsx
+import {
+  ThemeProvider,
+  createTheme,
+  experimental_sx as sx,
+} from '@mui/material/styles';
+
+<ThemeProvider
+  theme={createTheme({
+    components: {
+      MuiChip: {
+        styleOverrides: {
+          root: sx({
+            px: '12px', // shorthand for padding-left & right
+            py: '6px', // shorthand for padding-top & bottom
+            fontWeight: 500,
+            borderRadius: '8px',
+          }),
+          label: {
+            padding: 0,
+          },
+        },
+      },
+    },
+  })}
+>
+  ...your app
+</ThemeProvider>;
+```
+
+If you want to create a dynamic style based on props, you can return an array from the callback:
```

Comment:
Perhaps explain the example? Either: "In the following example" or with inline comments (or both). Also consider if other examples would benefit from further explanation.